### PR TITLE
Retain webview state when changing tabs

### PR DESCRIPTION
**Pull request by hunterwerlla**
- In views it makes sense to do
## Related issues
#1389 
#1257

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changes/next-release/Bug Fix-7a3b381b-15e3-4a93-a4e2-e27870f14261.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Bug Fix",
+	"description": "Retain view state for Step Functions and Lambda when chaging tabs"
```
Comment (justinmk3) [inline .changes/next-release/Bug Fix-7a3b381b-15e3-4a93-a4e2-e27870f14261.json:3]:
```suggestion
	"description": "Retain view state for Step Functions and Lambda when changing tabs"
```

**Review by justinmk3 — APPROVED** [review]
I have the impression that this approach is expensive or suboptimal, but it is definitely better to do this than block on the optimal approach.

**Comment by bryceitoc9** [comment]
MS recommends that we specifically don't do this: https://code.visualstudio.com/api/extension-guides/webview#retaincontextwhenhidden .

The proper way works if we build state into our webviews: https://code.visualstudio.com/api/extension-guides/webview#getstate-and-setstate ; this also has a benefit of being able to retain state between sessions: https://code.visualstudio.com/api/extension-guides/webview#serialization . I've got examples of this in my old `experiment/react*` branches.

If we merge this, we should definitely add a backlog item to do it the right way lest we use this as a crutch for all webviews going forward!

**Pull request merged**